### PR TITLE
Add a page describing Godot's release policy

### PR DESCRIPTION
**Pull request by Calinou**
This is a working document, feel free to suggest additions and changes.

### CODE REVIEW

File: about/release_policy.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+------------------------
+
+Godot versions are supported for a certain amount of time. While these durations
+are not set in stone, here's a table with the expected level of support
+for each Godot version:
+
++-------------+-------------------+---------------------------------------------------------------------------------------+
+| **Version** | **Release date**  | **Support level**                                                                     |
++-------------+-------------------+---------------------------------------------------------------------------------------+
+| Godot 4.0   | ~2021 (see below) | |partial| *Current focus of development (unstable).*                                  |
```
[the remark this answers is not in the document]
Comment (Calinou) [reply about/release_policy.rst:36]: `partial` is just the image name here (which is a yellow dot). I'm not sure if it's worth renaming it. Maybe I could add a gray dot image that would be `unstable` if you really want to.

File: about/release_policy.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+|eol| No support (end of life)
+
+Pre-release Godot versions aren't intended to be used in production and are
+provided on a best-effort basis.
+
+When is the next release out?
+-----------------------------
+
+While Godot contributors aren't working under any deadlines, there is usually
+a major release of Godot made available every year. Following this trend, this
```
[the remark this answers is not in the document]
Comment (Calinou) [reply about/release_policy.rst:70]: Indeed, I lumped minor and major releases together here. I guess I should make this more explicit.

File: about/release_policy.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+provided on a best-effort basis.
+
+When is the next release out?
+-----------------------------
+
+While Godot contributors aren't working under any deadlines, there's usually a
+major or minor Godot release made available every year. Following this trend,
+this means Godot 4.0 will most likely be released in **2021**.
+
+Patch releases are made available more frequently, typically every 2-6 months
```
Comment (akien-mga) [inline about/release_policy.rst:75]: 6 months is pretty extreme, I'd suggest `1-3 months` instead.

**Comment by akien-mga** [comment]
Thanks!

File: about/release_policy.rst (hunk 3 above)
Comment (aaronfranke) [reply about/release_policy.rst:75]: As of me writing this reply, it has been 4 months since the release of Godot 3.2.3, which is more than 3 months.

**Pull request merged**